src/gui/app/services/tts.service.js

```diff
@@ -92,11 +92,19 @@
                 msg.lang = 'en-US';
 
                 speechSynthesis.speak(msg);
+
+                const intMaxSoundLength = parseInt(maxSoundLength);
+                if (intMaxSoundLength > 0) {
+                    setTimeout(function() {
+                        speechSynthesis.cancel()
+                    }, maxSoundLength * 1000)
```

Comment:
If i'm reading this right, it looks like this timeout always fires for each tts that comes through that has a max sound length defined. Given that `speechSynthesis.cancel()` clears the current tts and the tts queue, this means we nuke the queue for each tts even if they dont actually surpass the length limit.

The solution for this is to clear the timeout when the given tts ends via [`SpeechSynthesisUtterance`'s "end" event](https://developer.mozilla.org/en-US/docs/Web/API/SpeechSynthesisUtterance/end_event). You should be able to do that like so:
```ts
const timeoutId = setTimeout(speechSynthesis.cancel, maxSoundLength * 1000);
msg.onend = () => {
   clearTimeout(timeoutId);
}
```
> Note: I've also simplified setTimeout by passing a reference to the speechSynthesis.cancel function instead of wrapping it in a new function.

Reply:
Ah, I didn't realize that all native TTS messages were handed off to the speech processor immediately and it managed the queue. Since there's no way to extract utterance boundaries or duration, I've removed the native TTS portion of this change. The only way that I could see that working would be to add a `Wait for the sound to finish before letting the next effect play`-like option to the native TTS option and turning the `read-tts` event into an async one but that seemed like a bit too much heavy lifting for what looked like a small change.